rosbag2/src/rosbag2/writer.cpp

```diff
@@ -63,7 +69,14 @@ void Writer::open(
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
+
   uri_ = storage_options.uri;
+
+  metadata_ = rosbag2_storage::BagMetadata{};
+  metadata_.storage_identifier = storage_->get_storage_identifier();
+  metadata_.starting_time = std::chrono::time_point<std::chrono::high_resolution_clock>(
+    std::chrono::nanoseconds::max());
+  metadata_.relative_file_paths.push_back(storage_->get_relative_path());
```

Comment:
this is one logical unit, consider creating a private function `initialize_metadata` to isolate this block.

Reply:
Applied suggestion